src/cogent3/core/new_alignment.py

```diff
@@ -2705,6 +2766,35 @@ def from_aligned_seqs(
             **kwargs,
         )
 
+    @classmethod
+    def from_seqs_and_gaps(
+        cls,
+        *,
+        seqs: dict[str, StrORBytesORArray],
+        gaps: dict[str, numpy.ndarray],
+        alphabet: new_alphabet.AlphabetABC,
+        **kwargs,
+    ):
+        """Construct an AlignedSeqsData object from a dict of ungapped sequences
+        and a corresponding dict of gap maps.
```

Comment:
change docstring so "gap maps" becomes "gap data". In the parameter definition, change `gap_maps` to `gaps` and indicate that col 0 is the sequence gap position and col 1 is the cumulative gap length